airbyte-cdk/python/unit_tests/sources/declarative/test_concurrent_declarative_source.py

```diff
@@ -615,27 +729,27 @@ def test_read_with_concurrent_and_synchronous_streams_with_concurrent_state():
         ),
     ]
 
-    source = ConcurrentDeclarativeSource(source_config=_MANIFEST, config=_CONFIG, catalog=_CATALOG, state=state)
+    concurrent_cursor_party_members_slices = [
+        {"start": "2024-07-01", "end": "2024-07-15"},
+        {"start": "2024-07-30", "end": "2024-08-13"},  # FIXME this is an interesting case where we restart from that top boundary. I'm wondering if we need to change that because if this was within one sync, we would start from 2024-09-01 and not 2024-07-30
```

Comment:
@brianjlai do you have an opinion on this? Basically, when we do sync without state, we would sync the following slices:
```
        {"start": "2024-07-01", "end": "2024-07-15"},
        {"start": "2024-07-16", "end": "2024-07-30"},
        {"start": "2024-07-31", "end": "2024-08-14"},
        {"start": "2024-08-15", "end": "2024-08-29"},
        {"start": "2024-08-30", "end": "2024-09-09"},
```

However when we start from a state where `{"start": "2024-07-16", "end": "2024-07-30"}` is part of the state, we will start from 2024-08-01 and not 2024-07-30. This was probably done to simplify the logic where this is necessary on the last slice but not the others. I think this concern is void once we start relying on `most_recent_cursor_value`.